cmd/common-main.go

```diff
@@ -443,6 +427,32 @@ func handleCommonCmdArgs(ctx *cli.Context) {
 	globalCertsCADir = &ConfigDir{path: filepath.Join(globalCertsDir.Get(), certsCADir)}
 
 	logger.FatalIf(mkdirAllIgnorePerm(globalCertsCADir.Get()), "Unable to create certs CA directory at %s", globalCertsCADir.Get())
+
+	// Check if we have configured a customer DNS cache TTL.
+	dnsTTL := ctx.GlobalDuration("dns-cache-ttl")
```

Comment:
GlobalDuration is not needed because dns-cache-ttl is not defined as a global flag (as other many flags), it is not needed you can drop it.

Reply:
that is a code to make sure if the flag gets moved around this is always honored, since this function is specifically for commonFlags()

Reply:
> that is a code to make sure if the flag gets moved around this is always honored, since this function is specifically for commonFlags()

It doesn't work @harshavardhana, I already tested it. For this to work, --dns-cache-ttls needs to be define when defining cli.App

Reply:
correct it is a defensive code